python/tests/test_logistic_regression.py

```diff
@@ -1257,3 +1276,258 @@ def test_functor(y: np.ndarray, err_msg: str) -> None:
+                label=1.0, weight=1.0, features=Vectors.dense([0.0, 1.0, 0.0])
+            ),  # (0., 1., 0.)
+            Row(
+                label=0.0, weight=1.0, features=Vectors.sparse(3, {0: 1.0})
+            ),  # (1., 0., 0.)
+            Row(
+                label=0.0, weight=1.0, features=Vectors.sparse(3, {0: 2.0, 2: -1.0})
+            ),  # (2., 0., -1.)
+        ]
+
+        bdf = spark.createDataFrame(data)
+
+        params: Dict[str, Any] = {
+            "regParam": 0.1,
+            "fitIntercept": fit_intercept,
+            "standardization": False,
+            "featuresCol": "features",
+            "labelCol": "label",
+        }
+
+        gpu_lr = LogisticRegression(**params)
+        assert gpu_lr.getOrDefault("enable_sparse_data_optim") == None
+        gpu_model = gpu_lr.fit(bdf)
+
+        cpu_lr = SparkLogisticRegression(**params)
+        cpu_model = cpu_lr.fit(bdf)
+        compare_model(gpu_model, cpu_model, bdf)
+
+
+@pytest.mark.compat
+@pytest.mark.parametrize("fit_intercept", [True, False])
+def test_compat_sparse_multinomial(
+    fit_intercept: bool,
+) -> None:
+    with CleanSparkSession() as spark:
+        data = [
+            Row(
+                label=1.0, weight=1.0, features=Vectors.sparse(3, {2: 1.0})
+            ),  # (0., 0., 1.)
+            Row(
+                label=1.0, weight=1.0, features=Vectors.sparse(3, {1: 1.0})
+            ),  # (0., 1., 0.)
+            Row(
+                label=0.0, weight=1.0, features=Vectors.sparse(3, {0: 1.0})
+            ),  # (1., 0., 0.)
+            Row(
+                label=2.0, weight=1.0, features=Vectors.sparse(3, {0: 2.0, 2: -1.0})
+            ),  # (2., 0., -1.)
+        ]
+
+        mdf = spark.createDataFrame(data)
+
+        params: Dict[str, Any] = {
+            "regParam": 0.1,
+            "fitIntercept": fit_intercept,
+            "standardization": False,
+            "featuresCol": "features",
+            "labelCol": "label",
+        }
+
+        gpu_lr = LogisticRegression(**params)
+        assert gpu_lr.getOrDefault("enable_sparse_data_optim") == None
+        gpu_model = gpu_lr.fit(mdf)
+
+        cpu_lr = SparkLogisticRegression(**params)
+        cpu_model = cpu_lr.fit(mdf)
+        compare_model(gpu_model, cpu_model, mdf)
+
+        for value in {True, False}:
+            gpu_lr = LogisticRegression(enable_sparse_data_optim=value, **params)
+            assert gpu_lr.getOrDefault("enable_sparse_data_optim") == value
+            gpu_model = gpu_lr.fit(mdf)
+            compare_model(gpu_model, cpu_model, mdf)
+
+
+@pytest.mark.parametrize("fit_intercept", [True, False])
+@pytest.mark.slow
+def test_sparse_nlp20news(
+    fit_intercept: bool,
+    caplog: LogCaptureFixture,
+) -> None:
+    datatype = np.float32
+    tolerance = 0.001
+    reg_param = 1e-6
+
+    from scipy.sparse import csr_matrix
+    from sklearn.datasets import fetch_20newsgroups
+    from sklearn.feature_extraction.text import CountVectorizer
```

Comment:
Would be nicer to use pyspark versions of CountVectorizer and train_test_split to add another example of api compatibility.

Reply:
We should use this example to add sparse version of our example jupyter notebook: https://github.com/NVIDIA/spark-rapids-ml/blob/branch-23.12/notebooks/logistic-regression.ipynb

Reply:
Replaced the sklearn CountVectorizer with Spark CountVectorizer and RegexTokenizer. 

Thinking to add the example to notebook in the next PR. It seems the NLP20NEWS accuracy and training iterations are sensitive to the tokenizer and hyperparameters (i.e. RegParam and maxIter). May need to spend sometime to understand the behavior.

Reply:
Sure.  Another PR for notebook is fine.